File: 231.power-of-two.java
/*
 * @lc app=leetcode id=231 lang=java
 *
 * [231] Power of Two
 */

// @lc code=start
class Solution {
    public boolean isPowerOfTwo(int n) {
        // Could you solve it without loops/recursion?
        // if one number, n is a power of 2,
        // then it will only have one 1 in binary code.
        return n > 0 && (n & (n - 1)) == 0;
    }
}
// @lc code=end

File: 231.power-of-two.py
#
# @lc app=leetcode id=231 lang=python3
#
# [231] Power of Two
#


# @lc code=start
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n <= 0:
            return False
        return n & (-n) == n


# @lc code=end
